rpi-u-boot-scr: Jump to preinit as required by read-only rootfs.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Joschka Seydell <joschka@seydell.org>

diff --git a/boot.cmd.in b/boot.cmd.in
index 61d0acf..de0cefb 100644
--- a/boot.cmd.in
+++ b/boot.cmd.in
@@ -31,7 +31,7 @@
 done
 
 if test -n "${bootpart}"; then
-  setenv bootargs "${bootargs} root=${bootpart} rauc.slot=${raucslot}"
+  setenv bootargs "${bootargs} root=${bootpart} rootfstype=squashfs init=/sbin/preinit rauc.slot=${raucslot}"
   saveenv
 else
   echo "No valid RAUC slot found. Resetting tries to 3"
@@ -41,6 +41,6 @@
   reset
 fi
 
-load ${BOOT_DEV} ${kernel_addr_r} boot/@@KERNEL_IMAGETYPE@@
+sqfsload ${BOOT_DEV} ${kernel_addr_r} /boot/@@KERNEL_IMAGETYPE@@
 if test ! -e mmc 0:1 uboot.env; then saveenv; fi;
 @@KERNEL_BOOTCMD@@ ${kernel_addr_r} - ${fdt_addr}
 